tests/main/mount-ns/task.yaml

```diff
@@ -0,0 +1,112 @@
+    snap pack test-snapd-mountinfo.core16
+    snap install --dangerous test-snapd-mountinfo_16_all.snap
+    snap connect test-snapd-mountinfo:mount-observe
+    su root -c "snap run test-snapd-mountinfo" >PER-SNAP-16.raw.txt
+    su test -c "snap run test-snapd-mountinfo" >PER-USER-16.raw.txt
+    sudo snap remove test-snapd-mountinfo
+
+    echo "Collect mountinfo from core18-based, per-snap and per-snap, per-suer mount namespaces"
+    snap pack test-snapd-mountinfo.core18
+    snap install --dangerous test-snapd-mountinfo_18_all.snap
+    snap connect test-snapd-mountinfo:mount-observe
+    su root -c "snap run test-snapd-mountinfo" >PER-SNAP-18.raw.txt
+    su test -c "snap run test-snapd-mountinfo" >PER-USER-18.raw.txt
+    sudo snap remove test-snapd-mountinfo
+
+    echo "Transform mountinfo tables to make them deterministic"
+    deterministic-mountinfo-tool                                              -f HOST.raw.txt        >HOST.deterministic.txt
+    deterministic-mountinfo-tool --ref HOST.raw.txt                           -f PER-SNAP-16.raw.txt >PER-SNAP-16.deterministic.txt
+    deterministic-mountinfo-tool --ref HOST.raw.txt --ref PER-SNAP-16.raw.txt -f PER-USER-16.raw.txt >PER-USER-16.deterministic.txt
+    deterministic-mountinfo-tool --ref HOST.raw.txt                           -f PER-SNAP-18.raw.txt >PER-SNAP-18.deterministic.txt
+    deterministic-mountinfo-tool --ref HOST.raw.txt --ref PER-SNAP-18.raw.txt -f PER-USER-18.raw.txt >PER-USER-18.deterministic.txt
+    if snap debug sandbox-features --required confinement-options:classic; then
+        deterministic-mountinfo-tool --ref HOST.raw.txt                           -f PER-SNAP-C7.raw.txt >PER-SNAP-C7.deterministic.txt
+        deterministic-mountinfo-tool --ref HOST.raw.txt --ref PER-SNAP-C7.raw.txt -f PER-USER-C7.raw.txt >PER-USER-C7.deterministic.txt
+    fi
+execute: |
+    diff -u "$SPREAD_BACKEND.$SPREAD_SYSTEM/HOST.expected.txt" HOST.deterministic.txt
+    diff -u "$SPREAD_BACKEND.$SPREAD_SYSTEM/PER-SNAP-16.expected.txt" PER-SNAP-16.deterministic.txt
+    diff -u "$SPREAD_BACKEND.$SPREAD_SYSTEM/PER-USER-16.expected.txt" PER-USER-16.deterministic.txt
+    diff -u "$SPREAD_BACKEND.$SPREAD_SYSTEM/PER-SNAP-18.expected.txt" PER-SNAP-18.deterministic.txt
+    diff -u "$SPREAD_BACKEND.$SPREAD_SYSTEM/PER-USER-18.expected.txt" PER-USER-18.deterministic.txt
+    if snap debug sandbox-features --required confinement-options:classic; then
+        diff -u "$SPREAD_BACKEND.$SPREAD_SYSTEM/PER-SNAP-C7.expected.txt" PER-SNAP-C7.deterministic.txt
+        diff -u "$SPREAD_BACKEND.$SPREAD_SYSTEM/PER-USER-C7.expected.txt" PER-USER-C7.deterministic.txt
+    fi
+restore: |
+    rm -f ./*.raw.txt ./*.deterministic.txt
+    rm -f test-snapd-mountinfo_c7_all.snap
+    rm -f test-snapd-mountinfo_16_all.snap
+    rm -f test-snapd-mountinfo_18_all.snap
+debug: |
+    for fname in ./*.deterministic.txt; do
```

Comment:
I fear this will generate too much noise and afaiu is not going to be super helpful. I think the failing diff output shoud be enough, no?

Reply:
This is actually more useful because you can take the full file and inspect it offline with tools better than just diff alone. It is also very useful for creating new data sets for things you don't have locally.

I agree it will be noisy if this fails often but I think we can reconsider after the initial painful phase.

Reply:
Ok, I trust you on this since you've been dealing with these failures while working on the test. My main point is we're outputing here all of them, while showing the one that failed would be enough, right? Perhaps we should simply `rm -rf <name>.deterministic.txt`after every diff step, so we are immediately getting rid of sucesful ones and not show then in debug on failure?

Reply:
I would say that the interesting one is the HOST as that will show cruft leaking into the mount namespace. Non-determinism of the remaining ones seems to be handled well now and any differences there will be reflections of what HOST differs.

As I said, I'm happy to evolve this but I would keep it as-is for now, because having easy access to this even without `-debug` is very useful. I can change how the debug section works after the qemu data set is merged.